src/components/common/ShareButton/ShareButton.tsx

```diff
@@ -0,0 +1,37 @@
+import Image from 'next/image'
+import React from 'react'
+import { AiOutlineCopy } from 'react-icons/ai'
+import { toast, ToastContainer } from 'react-toastify'
+import { ShareButtonProps } from './ShareButton.type'
+import 'react-toastify/dist/ReactToastify.css'
+
+const ShareButton: React.FC<ShareButtonProps> = () => {
+  const handleCopy = async () => {
+    try {
+      await navigator.clipboard.writeText(location.href)
+      toast.success('URLをコピーしました')
+    } catch (error) {
+      alert('URLをコピーできませんでした')
+    }
+  }
+
+  const handleShareLine = () => {
+    // Lineに共有するURLを作成
+    const lineShareUrl = `https://social-plugins.line.me/lineit/share?url=${location.href}`
+    window.open(lineShareUrl, '_blank') // 新しいタブで開く
+  }
+
+  return (
+    <div className='flex items-center'>
+      <button className='btn btn-ghost' onClick={handleCopy}>
+        <AiOutlineCopy size='25' />
+      </button>
```

Comment:
リンクコピーボタンは丸くしちゃっていいんじゃないかと思うのと、gapをちょっとだけつけてもいいと思う

```suggestion
    <div className='flex items-center gap-2'>
      <button className='btn btn-circle btn-ghost' onClick={handleCopy}>
        <AiOutlineCopy size='25' />
      </button>
```

Reply:
[d7d8459](https://github.com/NUTFes/chosei-chan/pull/32/commits/d7d845989f510b42d70b9048542acf9bb89e87bf)